h2o-core/src/main/java/water/H2O.java

```diff
@@ -1965,6 +1992,49 @@ public void run() {
     }
   }
 
+  /**
+   * Register {@link NodeDisconnectHook}
+   * @param hook to register
+   */
+  public static void registerNodeDisconnectedHook(NodeDisconnectHook hook){
+    nodeDisconnectHooks.add(hook);
+  }
+
+  /**
+   *   Thread used to run disconnect hooks on nodes who disconnects from the cloud
+   */
+  public static class DisconnectedNodesWatchdog extends Thread {
+    final private int sleepMillis = 6000; // 6 seconds
```

Comment:
You can make it configurable via JVM property `-D` (look for SYSTEM_PROP_PREFIX)